snappy/snap_remote_repo.go

```diff
@@ -230,10 +229,20 @@ func (s *SnapUbuntuStoreRepository) Details(name string, origin string) (parts [
 		return nil, err
 	}
 
-	snap := NewRemoteSnapPart(detailsData)
-	parts = append(parts, snap)
+	return NewRemoteSnapPart(detailsData), nil
+}
 
-	return parts, nil
+// Details returns details for the given snap in this repository
+func (s *SnapUbuntuStoreRepository) Details(name string, origin string) (parts []Part, err error) {
+	snapName := name
```

Comment:
What's the advantage here?  Just use name below (`name = ...`)?

It also makes me wonder if this function has much of a reason to exist. Feels like Snap itself could be used anywhere we have it, but might be missing the larger context.